lnrpc/signrpc/signer.proto

```diff
@@ -220,10 +221,11 @@ message SharedKeyRequest {
     bytes ephemeral_pubkey = 1;
 
     /*
-    The optional key locator of the local key that should be used. If this
-    parameter is not set then the node's identity private key will be used.
+    A key descriptor describes the key used for performing ECDH. The key_family
+    of the key locator must be supplied. Then either a raw public key or a
+    key_index of the key locator should be passed, but not both.
     */
-    KeyLocator key_loc = 2;
+    KeyDescriptor key_desc = 2;
```

Comment:
There seems to be a misunderstanding here, sorry. We cannot just change the type of an existing (=released) RPC field. Adding a new field (with the number `3`) and deprecating the old one is the correct way to go.
But as long as the old field is still in the proto (even when declared with `[deprecated = true]`), it needs to be evaluated (which means, check if old field is set, if not check new field, throw an error if both fields are set).
Only once we completely remove a field (after it's been deprecated for at least 1 full release cycle), we set it to `reserved = 2;` and remove the logic that reads its value.

So we should leave `KeyLocator key_loc = 2` but add `[deprecated = true]` to it. Then add `KeyDescriptor key_desc = 3;` as the new preferred field.